NAS-117236 / 22.12-RC.1 / Check for keyboard support on new storage pages

AlexKarpov98 opened the pull request:
checked accessibility via keyboard, made sure everything highlights and responds to user keyboard actions.

<img width="289" alt="Screen Shot 2022-10-20 at 13 01 00" src="https://user-images.githubusercontent.com/22980553/196919209-1d3d71b1-fa43-420b-b313-155906bd4ca6.png">

bugclerk commented:
Jira URL: https://ixsystems.atlassian.net/browse/NAS-117236

undsoft reviewed (changes requested):
Cannot select device in Devices, cannot select a dataset in Datasets.

AlexKarpov98 commented:
Fixed hovering effect for **tree-item**
<img width="676" alt="Screen Shot 2022-10-21 at 15 19 12" src="https://user-images.githubusercontent.com/22980553/197194060-8d0a0ea8-eb44-43ec-8b23-bea6d72e3708.png">

And added **enter press** support to view tree-item's data.

undsoft commented on src/app/pages/datasets/components/dataset-management/dataset-management.component.html:
```diff
@@ -86,6 +87,7 @@
                 routerLinkActive="selected"
                 [routerLink]="['/datasets', dataset.id]"
                 (click)="openMobileDetails()"
+                (keydown.enter)="openMobileDetails(['/datasets', dataset.id])"
```
It looks like you can unify all of this: 
1. Remove routerLink directive.
2. Have the same handler for (click) and (keydown.enter).
3. Call it something else like `onRowSelected(dataset: Dataset)`

undsoft reviewed (approved):
If I am in a smaller screen and have mobile layout, it seems like focus stays in the table and it's hard to start using elements in the mobile view (I have to Tab through the whole table without seeing anything happening).
See if there is an easy fix for this. If there is no simple fix, you can merge without it.

AlexKarpov98 commented:
@undsoft - have a look at it now please :)

undsoft reviewed (approved):
There is a button to expand/collapse rows. It is no longer possible to expand row on mobile without going to mobile details.
That is if you have a collapsed row and you want to select a child, it's not going to be easy.

Otherwise works okay.

AlexKarpov98 commented:
tested and now works correctly

undsoft reviewed (approved):
Works okay.
I would do
```
(keydown.enter)="$event.stopPropagation()"
```
on button instead of looking at element tag. Less fragile.
